### PR TITLE
Update max request limit and number of processing threads

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by huffman** [comment]
I'm in the process of doing more testing, albeit on a single machine, but this may inform some slight changes in these numbers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 489d2e4ca50ecc3ecc3ece28304c352b4352a472 available at: https://gist.github.com/e12a2e859a8c635aaca3c38ffa80bac9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8216.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8216.dmg

**Comment by cozza13** [comment]
Smoke test

**Comment by OmegaHeron** [comment]
Will this be exposed to a setting variable at some point?  I'd like to experiment with this...  it might solve some of my issues but I'm not setup to compile on this machine and it would take all day to change, recompile, test - repeat.

My machine chokes on loading standard home, it seems, and I stress seems, due to getting more work during rez phase than it can handle.  Hackish methods to throttle ATP points to that being somewhat valid and this might offer a way to better manage that.

**Comment by cozza13** [comment]
@OmegaHeron , @huffman could know about exposing it. Be great if you could test it

**Comment by OmegaHeron** [comment]
As a general smoke test - it behaves exactly as current production code - scenes load, everything works.

**Comment by cozza13** [comment]
https://trello.com/c/OJFvFXin/6-increase-number-of-concurrent-downloads

**Comment by huffman** [comment]
@OmegaHeron Would it be enough to have it available as something you can set in your settings file, but not exposed in the gui?

**Comment by huffman** [comment]
(Keeping in mind that you would need to know the name of the setting to set it)

**Comment by OmegaHeron** [comment]
Yes @huffman either a manual add of two params to ini or env vars is fine if you're so inclined to make available.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -517,7 +520,7 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer) :
     // (main thread, present thread, random OS load)
     // More threads == faster concurrent loads, but also more concurrent
     // load on the GPU until we can serialize GPU transfers (off the main thread)
-    QThreadPool::globalInstance()->setMaxThreadCount(2);
+    QThreadPool::globalInstance()->setMaxThreadCount(PROCESSING_THREAD_POOL_SIZE);
```
Comment (jherico) [inline interface/src/Application.cpp]: The number of threads dedicated to network traffic should probably be partially derived from the number of cores on the machine.  While 2 is conservative, it does improve performance on lower end hardware and hardware with less effective task switching like the i5.  We likely need to ensure that at least 1 or possibly 2 cores are reserved for the main thread and the presentation thread.

Comment (ZappoMan) [inline interface/src/Application.cpp]: I agree with @jherico - we can't just bump this to 6 -- we need something like:
```
int MIN_THREADS = 2; // what we used to have
int RESERVED_CORES = 2; // one for present, one for main thread
auto threadCount = QThread::idealThreadCount() > RESERVED_CORES + MIN_THREADS ? QThread::idealThreadCount() - RESERVED_CORES  : MIN_THREADS;
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: fcd59ea94f122c8644bf5a95cc7246cbf5ff382d available at: https://gist.github.com/e12a2e859a8c635aaca3c38ffa80bac9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8216-fcd59ea.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8216-fcd59ea.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10 - Full smoke Test
i5 6500 CPU - GTX970 GPU - Desktop and DK2 HMD tested.

## QA Test
**QA Tester**: OmegaHeron
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** FAILURE
**playa 'playaPerformanceTest.js' Result**: **playa 'playaPerformanceTest.js'** FAILURE
**cellscience 'playaPerformanceTest.js' Result**: **cellscience 'playaPerformanceTest.js'** FAILURE
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:

In reference to playaPerforanceTest items marked as fails.

playaPerformanceTest.js results **will** pass on dev-playa and dev-demo **if** I have not visited localhost or any other "heavy" domains where Interface has a lot of GPU memory shuffling to do when coming to demo or playa.  This is **not new and is not made worse/better by this PR compared to DEV/RC builds**.  localhost continues to be same state as my last analysis - totally unusable with Interface and Stack running on same machine such as mine.  No change there.  If I'm willing to wait 5(ish) minutes I can ultimately load localhost to a stable state if I stand still - wait, rotate a tiny bit, wait and repeat until I've made a 360 degree rotation.  Then it's fine.  That's how I managed to get stats.

Desktop render stats

**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |60 |60 |60
**Present Rate**: |60 |60 |60
**Present New Rate**: |60 |60 |60
**CPU Buffer Count**: |5202 |5467 |4972
**GPU Buffer Count**: |1662 |1842 |1173
**CPU Memory Usage**: |78565376 |91660288 |104599552
**GPU Memory Usage**: |57151488 |65146880 |59502592
**CPU Texture Count**: |307 |392 |328
**GPU Texture Count**: |528 |950 |1037
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |80119012 |414509760 |315595408
**GPU Texture Memory Usage**: |1109999662 |1912639263 |1259483737
**GPU Virtual Memory Usage**: |1191469102|2003693343|1360122457
**Triangle Count**: |217953 |345471 |466208
**Triangle Rate**: |13622062 |21591937 |29138000
**API Drawcall Count**: |169 |106 |133
**Drawcall Count**: |300 |107 |336
**Drawcall Rate**: |18750 |6687 |21000
**State Changes Texture Count**:|147 |144 |200
**Pipeline Count**: |30 |22 |29
**Item Opaques**: |188 |156 |118
**Item Translucents**: |31 |3 |26
**Item Lights**: |17 |18 |3
**Timing Opaques**: |966 |1167 |1048
**Timing Translucents**: |218 |78 |306
**Timing Lighting**: |141 |162 |89
**Render Frame**: |2054 |2246 |2116

DK2 render stats

**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |75 |75 |75
**Present Rate**: |75 |75 |75
**Present New Rate**: |75 |75 |75
**CPU Buffer Count**: |6085 |5826 |5302
**GPU Buffer Count**: |2087 |2176 |1579
**CPU Memory Usage**: |100524032 |102813696 |105304064
**GPU Memory Usage**: |72474624 |74022912 |59633664
**CPU Texture Count**: |356 |440 |360
**GPU Texture Count**: |1577 |1059 |1163
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |80611652 |172118232 |242817488
**GPU Texture Memory Usage**: |1249556805 |2283515572 |1259855063
**GPU Virtual Memory Usage**: |1510480197|3160389172|2136728663
**Triangle Count**: |392654 |1636165 |1478896
**Triangle Rate**: |24540875 |125858846 |113761230
**API Drawcall Count**: |208 |310 |318
**Drawcall Count**: |263 |641 |2771
**Drawcall Rate**: |16437 |49307 |213153
**State Changes Texture Count**:|196 |308 |350
**Pipeline Count**: |24 |37 |48
**Item Opaques**: |237 |389 |301
**Item Translucents**: |10 |29 |71
**Item Lights**: |20 |45 |26
**Timing Opaques**: |1345 |2436 |2235
**Timing Translucents**: |122 |218 |457
**Timing Lighting**: |162 |285 |258
**Render Frame**: |2722 |4117 |3896

Some other notes - I **absolutely have to set minimum FPS to 0 for desktop and hmd** -- If I do not then when my FPS tanks it goes into LOD action that worsens problem as it constantly cycles between lowering/raising LOD value as scene loads.  As can be seen from my stats -- *if* I can get home scene loaded and stable - FPS is not an issue.  Again - this is not new or this PR's problem.

Another oddity I'm seeing - almost certainly not from this PR is connection issues.  Specifically, if I've connected to domains using full auto networking I seem to have problems connecting to a domain that is **not** using automatic networking.  For instance, I can initially start at Heron (manual networking, no ICE required) - move to dev-playa, dev-demo and localhost but any attempt to return to heron results in 1) no connection or 2) connection but only some random number of AC servers are connected to.  If I restart Interface it connects to Heron fine.  If I begin at dev-playa, dev-demo or localhost I have yet to be able to connect to Heron.  I've not seen this behavior in any other PR/DEV build, but, will be checking for it in others next.

**Comment by ZappoMan** [comment]
@OmegaHeron - re: "Another oddity I'm seeing - if I've connected to domains using full auto networking I seem to have problems connecting to a domain that is not using automatic networking. " --- Can you confirm you are seeing this behavior in the latest `dev-download` build?

**Comment by OmegaHeron** [comment]
@zappoman - that's what I'm checking now.

**Comment by OmegaHeron** [comment]
@ZappoMan - seeing same with 5140.  Trying 5130 on a hunch.

**Comment by OmegaHeron** [comment]
Drop red alert for now @zappoman - It **may** be a network issue with Digital Ocean's NYC data center.  Will continue to monitor and advise if it seems to be not network related/my problem.

**Pull request merged**